crypto/x509/x509_d2.c

```diff
@@ -27,10 +27,14 @@ int X509_STORE_set_default_paths_ex(X509_STORE *ctx, OSSL_LIB_CTX *libctx,
         return 0;
     X509_LOOKUP_add_dir(lookup, NULL, X509_FILETYPE_DEFAULT);
 
+    /*
+     * Also search the default store URIs, if any, but
+     * DO NOT add the default search directory also to this lookup,
+     * such a second search would be slower and redundant.
+     */
     lookup = X509_STORE_add_lookup(ctx, X509_LOOKUP_store());
     if (lookup == NULL)
         return 0;
-    X509_LOOKUP_add_store_ex(lookup, NULL, libctx, propq);
```

Comment:
If we some day have a default URI, presumably we'd add a call to instantiate that, rather than this call?
Though, yes, it is a bit odd that the default directory is silently added when a NULL source is supplied, that case should be a NOP, or maybe even return an error indication about the missing argument.

[ I did read the code you linked to by the way, and decided to leave it alone for now, in case some existing code, perhaps even outside OpenSSL, relies on its current behaviour. ]

Reply:
> If we some day have a default URI, presumably we'd add a call to instantiate that, rather than this call?

Er, isn't this function the generic call to instantiate all defaults?  Why would a STORE URI be excluded?

But otherwise, there's a reason I'm of two minds here...  STORE URIs have the `file:` scheme as implicit default, i.e. you can give it any file path and it is able to use that (by loading an implementation of the `file:` scheme, which we have built in).  So conceptually speaking, a directory *is a* URI and is perfectly usable with the STORE API.

However, thinking about this further, it may be questionable to use the *built in default directory* as a built in default URI, so all things considered, I'm still convinced that changing `crypto/x509/by_store.c` is a better move...  and at least for the moment, I'd say that the `X509_L_ADD_STORE` handler could simply return 1 if `argp == NULL`.

Reply:
I had actually considered the option you suggested, but was hesitant to go there at the time. The result is I think cleaner (no kluky magic with the default directory). So I'm happy you see that as the right way forward. The call I deleted now becomes a NOP, but perhaps some day some other (non-duplicative) default might result from that call?

If calls with argp == NULL will always just short-circuit to NOP, why make them?  That is, after simplifying the callee, should *still* remove the call, or if not, why not?

Reply:
I would keep that call, because like you say, there may be a default that gets inserted into `crypto/x509/by_store.c` some day...  will we remember to add back the line this PR removes?  History shows that we're [ahem] collectively a bit prone to forget that sort of thing...

Reply:
Done.